modules/Microsoft.EventGrid/domainTopics/.test/common/dependencies.bicep

```diff
@@ -0,0 +1,16 @@
+@description('Optional. The location to deploy to.')
+param location string = resourceGroup().location
+
+@description('Required. The name of the Event Grid Domain to create.')
+param eventGridDomainname string
+
+resource domain 'Microsoft.EventGrid/domains@2022-06-15' = {
```

Comment:
Why calling the module domainTopics if the resource is domains? I'd suggest to rename the module folder to `domains` and align all corresponding occurrences accordingly

Reply:
Domain resource is dependency for testing the topic resource. We have already domains module. This module is required if you want to create topics under Event Grid Domains.

Reply:
Ok, then we need to add the topics module as a child of the domains module. This means in practice:
- add a subfolder `topics` to `eventgrid/domains`
- move the `topics` module there including the topics template, its nested role assignment, its version, its readme
- remove the added ado and gh pipelines, as in CARML we are adding them only for parent level modules (domains in this case which is already existing)
- extend the domains deploy.bicep template to optionally deploy the topics child module. Similarly to how e.g., a keyvault can deploy keyvault keys.
- extend the domains common test to also deploy domain topics
- regenerate readme files for both domain/topics and domains modules